discovery/seata-discovery-raft/src/main/java/io/seata/discovery/registry/raft/RaftRegistryServiceImpl.java

```diff
@@ -339,8 +403,49 @@ private static void acquireClusterMetaData(String clusterName, String group) {
         }
     }
 
+    private static void refreshToken() {
+        // if username and password is not in config , return
+        String username = CONFIG.getConfig(getRaftUserNameKey());
+        String password = CONFIG.getConfig(getRaftPassWordKey());
+        if (Objects.isNull(username) || Objects.isNull(password)) {
+            return;
+        }
+        String raftClusterAddress = CONFIG.getConfig(getRaftAddrFileKey());
+        // get token and set it in cache
+        if (StringUtils.isNotBlank(raftClusterAddress)) {
+            String[] tcAddressList = raftClusterAddress.split(",");
+            String tcAddress = tcAddressList[0];
+            Map<String, String> param = new HashMap<>();
+            param.put(PRO_USERNAME_KEY, username);
+            param.put(PRO_PASSWORD_KEY, password);
+            String response = null;
+            try (CloseableHttpResponse httpResponse =
+                     doPost("http://" + tcAddress + "/api/v1/auth/login", param, null, 1000, "application/json")) {
+                if (httpResponse != null) {
+                    if (httpResponse.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
+                        response = EntityUtils.toString(httpResponse.getEntity(), StandardCharsets.UTF_8);
+                        JsonNode jsonNode = OBJECT_MAPPER.readTree(response);
+                        String codeStatus = jsonNode.get("code").asText();
+                        if (!StringUtils.equals(codeStatus, "200")) {
+                            //authorized failed,throw exception to kill process
+                            throw new AuthenticationFailedException("Authentication failed! you should configure the correct username and password.");
+                        }
+                        jwtToken = jsonNode.get("data").asText();
+                        tokenTimeStamp = System.currentTimeMillis();
```

Comment:
tokenTimeStamp放到发送请求之前更好,因为网络是有时延的,更早的刷新token更安全
It's better to put the tokenTimeStamp before sending the request, because the network has a time delay and it's safer to refresh the token earlier.

Reply:
这样的话是不是会有以下场景：
如果先刷新tokenTimeStamp，然后因为网络原因发送请求超时了，超时异常被捕获并打日志，可能会出现tokenTimestamp刷新了，但是token还是旧的情况，客户端下一次请求带着旧的token访问server被拒绝，返回401，客户端抛出异常停止线程。
In this scenario, if the tokenTimeStamp is refreshed first and then a request times out due to network issues, with the timeout exception being caught and logged, there is a possibility of the tokenTimestamp being updated while the actual token remains outdated. Subsequently, when the client attempts the next request with the old token, the server rejects the request, responding with a 401 status code. As a result, the client throws an exception, halting the thread.

Reply:
fixed